source/fundamentals/class-mapping.txt

```diff
@@ -0,0 +1,328 @@
+map to the ``Name`` property and the ``age`` argument will map to the ``Age`` property.
+
+.. code-block:: csharp
+   :emphasize-lines: 7
+
+   public class Person 
+   {
+       public string Name { get; set; }
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+
+       [BsonConstructor]
+       public Person(string name, string age)
+       {
+           Name = name;
+           Age = age;
+       }
+   }
+
+.. tip:: Multiple ``BsonConstructor`` attributes
+
+   If there is more than one constructor with the ``BsonConstructor``
+   attribute, the driver uses the constructor that has the most
+   parameters with matching fields in the document.
+
+You can also specify the constructor to use when registering your class map:
+
+.. code-block:: csharp
+   :emphasize-lines: 17
+
+   public class Person 
+   {
+       public string Name { get; set; }
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+
+       public Person(string name, string age)
+       {
+           Name = name;
+           Age = age;
+       }
+   }
+
+   BsonClassMap.RegisterClassMap<Person>(classmap =>
+   {
+       classmap.AutoMap();
+       classmap.MapCreator(p => new Person(p.Name, p.Age));
+   });
+
+Customize Property Serialization
+--------------------------------
+
+You can customize how the driver serializes a class property by
+adding attributes to the property. For more information about custom
+property serialization, see :ref:`csharp-custom-serialization`.
+
+Support Extra Elements
+~~~~~~~~~~~~~~~~~~~~~~
+
+You can design your {+language+} class to store any extra elements in your
+document that don't have matching class properties. To do this your class must
+have a ``BsonDocument`` type property to hold the extra elements.
+
+The following code uses the ``BsonExtraElements`` attribute with the
+``ExtraElements`` property to direct the driver to store extra elements:
+
+.. code-block:: csharp
+
+   public class Person
+   {
+       public string Name { get; set; 
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+       [BsonExtraElements]
+       public BsonDocument ExtraElements {get; set;}
+   }
+
+You can also support extra elements when initializing a class map as follows:
+
+.. code-block:: csharp
+   
+   BsonClassMap.RegisterClassMap<Person>(classmap => 
+   {
+       classmap.AutoMap();
+       classmap.MapExtraElementsMember(p => p.ExtraElements);
+   });
+
+.. note::
+
+   After the driver serializes a class with extra elements back to BSON, the 
+   extra elements may not be in the same order as they were in the original 
+   document.
+
+Dynamically Serialize Properties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use a method to determine whether or not to serialize a property. For
+the driver to automatically use the method when serializing, the method must be
+named ``ShouldSerialize<property name>``. When the driver sees a method with
```

Comment:
Angle brackets used like this could be confusing in C# because that looks like a generic method.

Not sure if there is a better alternative though.